metacat-common-server/src/main/java/com/netflix/metacat/common/server/events/MetacatEventBus.java

```diff
@@ -79,22 +77,29 @@ public void postAsync(final ApplicationEvent event) {
     public void postSync(final ApplicationEvent event) {
         log.debug("Received request to post an event {} synchronously", event);
         this.eventSyncCounter.increment();
-        this.eventPublisher.publishEvent(event);
+        super.multicastEvent(event);
     }
 
-    /**
-     * Registers an object.
-     *
-     * @param object object
-     */
-    public void register(final Object object) {
+    @Override
+    public void addApplicationListener(final ApplicationListener listener) {
+        super.addApplicationListener(listener);
+        asyncEventMulticaster.addApplicationListener(listener);
     }
 
-    /**
-     * De-registers an object.
-     *
-     * @param object object
-     */
-    public void unregister(final Object object) {
+    @Override
+    public void removeApplicationListener(final ApplicationListener listener) {
```

Comment:
You should at least put comments as to why you're doing this for future reference

Reply:
Will do.